### PR TITLE
NEW: Dualshock3 support on Desktops

### PR DESCRIPTION
**Pull request by jechter**
Adds support for DualShock 3 gamepads on Desktops via HID. Only supports Input, no rumble - the DS3 gamepad I have for testing  is a Sixaxis controller, which has no rumble capabilities.

### CODE REVIEW

File: Assets/Tests/InputSystem/Plugins/DualShockTests.cs [hunk 1]
```diff
@@ -78,6 +78,67 @@ public void Devices_SupportsDualShockAsHID()
         // Sensors not (yet?) supported. Needs figuring out how to interpret the HID data.
     }
 
+    [Test]
+    [Category("Devices")]
+    public void Devices_SupportsDualShock3AsHID()
```
Comment (Rene-Damm) [inline Assets/Tests/InputSystem/Plugins/DualShockTests.cs]: Test looks identical? Recommend just using `TestCase` in that case.

File: Packages/com.unity.inputsystem/InputSystem/Plugins/DualShock/DualShockGamepadHID.cs [hunk 2]
```diff
@@ -77,6 +77,68 @@ public FourCC format
         }
     }
 
+    /// <summary>
+    /// Structure of HID input reports for PS3 DualShock 3 controllers.
+    /// </summary>
+    [StructLayout(LayoutKind.Explicit, Size = 32)]
+    public struct DualShock3HIDInputReport : IInputStateTypeInfo
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/DualShock/DualShockGamepadHID.cs:84]: How different are the two? Any chance we can just have the same struct and use a layout variant?

File: Assets/Tests/InputSystem/Plugins/DualShockTests.cs (hunk 1 above)
Comment (jechter) [reply Assets/Tests/InputSystem/Plugins/DualShockTests.cs]: Differences are in types, not suitable for testcase. But I was able to get rid of the code duplication by making a shared generic method for the test code.

File: Packages/com.unity.inputsystem/InputSystem/Plugins/DualShock/DualShockGamepadHID.cs (hunk 2 above)
Comment (jechter) [reply Packages/com.unity.inputsystem/InputSystem/Plugins/DualShock/DualShockGamepadHID.cs:84]: Nope. 100% different in binary layout.

**Review by Rene-Damm — APPROVED** [review]

**Review by douglas-piconi — APPROVED** [review]

**Pull request merged**